src/execution/sql/table_vector_iterator.cpp

```diff
@@ -94,6 +97,19 @@ class ScanTask {
         thread_state_container_(exec_ctx->GetThreadStateContainer()),
         scanner_(scanner) {}
 
+  ScanTask(uint32_t table_oid, uint32_t *col_oids, uint32_t num_oids, void *const query_state,
```

Comment:
Instead of modifying ScanTask, can you split changes into your own CreateIndexTask?